File: CoNote.UI/src/api/Component/models/CreateComponentRequest.ts
import { ComponentType } from "../../../models/enums/ComponentType";
import { StyleProperties } from "../../../models/views/ComponentView";

export interface CreateComponentRequest {
  worksheetId: number;
  width: number;
  height: number;
  zIndex: number;
  x: number;
  y: number;
  rotation: number;
  type: ComponentType;
  content?: string;
  style?: StyleProperties;
}

File: CoNote.Services/Components/Models/CreateComponentRequest.cs
using CoNote.Core.Enums;
using CoNote.Core.Models;

namespace CoNote.Services.Components.Models;
public class CreateComponentRequest
{
    public long WorksheetId { get; set; }
    public int Width { get; set; }
    public int Height { get; set; }
    public int ZIndex { get; set; } = 1;
    public double X { get; set; }
    public double Y { get; set; }
    public double Rotation { get; set; } = 0;
    public ComponentType Type { get; set; }
    public string? Content { get; set; }
    public StyleProperties? Style { get; set; }
}